packages/utils/src/misc.ts

```diff
@@ -321,3 +322,43 @@ export function engineEvent(event: EngineTypes.Event, id?: number | string | und
 export function mergeArrays<T>(a: T[] = [], b: T[] = []): T[] {
   return [...new Set([...a, ...b])];
 }
+
+export async function handleDeeplinkRedirect({
+  id,
+  topic,
+  store,
+}: {
+  id: number;
+  topic: string;
+  store: IKeyValueStorage;
```

Comment:
```suggestion
  deepLink: string;
```
I think if we're to make this general util, that others can use too, might be better to pass the `deepLink` directly instead of `store`. This way, the util would work with any deep link and not be locked for `WALLETCONNECT_DEEPLINK_CHOICE` only. Wdyt?